addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -3702,6 +3702,49 @@ QUnit.module('relational_fields', {
 
         form.destroy();
     });
+    QUnit.test('Check onchange with two consecutive many2one', async function (assert) {
+        assert.expect(2);
+        this.data.product.fields.product_partner_ids = { string: "User", type: 'one2many', relation: 'partner' };
+        this.data.product.records[0].product_partner_ids = [1];
+        this.data.product.records[1].product_partner_ids = [2];
+        this.data.turtle.fields.product_ids = { string: "Product", type: "one2many", relation: 'product' };
+        this.data.turtle.fields.user_ids = { string: "Product", type: "one2many", relation: 'user' };
+        this.data.turtle.onchanges = {
+            turtle_trululu: function (record) {
+                record.product_ids = [37];
+                record.user_ids = [17, 19];
+            },
+        };
+        var form = await createView({
+            View: FormView,
+            model: 'turtle',
+            data: this.data,
+            arch: 
+                '<form string="Turtles">' +
+                    '<field string="Product" name="turtle_trululu"/>' +
+                    '<field readonly="1" string="Related field" name="product_ids">' +
+                        '<tree>' +
+                            '<field widget="many2many_tags" name="product_partner_ids"/>' +
+                        '</tree>' +
+                    '</field>' +
+                    '<field readonly="1" string="Second related field" name="user_ids">' +
+                        '<tree>' +
+                            '<field widget="many2many_tags" name="partner_ids"/>' +
+                        '</tree>' +
+                    '</field>' +
+                '</form>',
+            res_id: 1,
+        });
+
+        await testUtils.form.clickEdit(form);
+        await testUtils.fields.many2one.clickOpenDropdown("turtle_trululu");
+        await testUtils.fields.many2one.searchAndClickItem('turtle_trululu', {search: 'first record'});
+        assert.containsN(form, '.o_field_many2manytags[name="product_partner_ids"] .badge.o_tag_color_0', 1,
+            "one tag should be present");
+        assert.containsN(form, '.o_field_many2manytags[name="partner_ids"] .badge.o_tag_color_0', 2,
+            "two tags should be present");
```

Comment:
you need to check that the display names are correct (before your fix, we didn't run the function that fetches them for the first one)